exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/metric_exporter/__init__.py

```diff
@@ -175,13 +176,27 @@ def _export(self, serialized_data: bytes):
         elif self._compression == Compression.Deflate:
             data = zlib.compress(serialized_data)
 
-        return self._session.post(
```

Comment:
Hey 👋

Got here from https://github.com/open-telemetry/opentelemetry-python/issues/4476 affecting us at $workplace.

Would using `requests` support for urllib3-based automatic retries[^1]?

For example, to declare that the first connection error[^2] will be retried

```python
from requests.adapters import HTTPAdapter
from urllib3.util import Retry

class OTLPMetricExporter(MetricExporter, OTLPMetricExporterMixin):
    def __init__():
        ...
        self._session = session or requests.Session()
        self._session.mount("https://", HTTPAdapter(max_retries=Retry(connect=1)))
```

[^1]: https://requests.readthedocs.io/en/latest/user/advanced/#example-automatic-retries
[^2]: https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html#urllib3.util.Retry

Reply:
That looks like a nice cleanup

Reply:
thanks for the suggestion @edgarrmondragon, I tested it but am still getting:

```
File "./.venv/lib/python3.11/site-packages/requests/adapters.py", line 682, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```

After reading the code in adapters.py and in the underlying urllib, it doesn't look like connection errors fall into the category of using the retries

Reply:
Ah, gotcha. Since `requests.exceptions.ConnectionError` is not an exception that `urllib3.util.Retry` recognizes as a connection error (those only being `urllib3.exceptions.ProxyError` and `urllib3.exceptions.ConnectTimeoutError`) it's not handled.

That means you'd need to use `Retry(total=...)` or `Retry(other=...)` to make sure those `requests` exceptions are retried.

Reply:
Thanks for the input @edgarrmondragon.

I tested this with `total` and `other` as well, but couldn't get this to work without the try/except block in my PR. I'm marking this comment as resolved for now, if there's a better workaround in the future, I would be happy to review a PR for it